packages/atclient/tests/test_atserver_message.c

```diff
@@ -0,0 +1,245 @@
+#include "atlogger/atlogger.h"
+#include "atserver_message.h"
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define BASE_TAG(X) "test_atserver_message " X
+
+// non-heap tests
+static uint8_t test_1a_long_prompt();
+static uint8_t test_1b_short_prompt();
+static uint8_t test_1c_no_prompt();
+static uint8_t test_2a_no_token();
+static uint8_t test_3a_no_body();
+static uint8_t test_4a_empty_message();
+
+// heap tests
+static uint8_t test_5a_heap();
+static uint8_t test_5b_bad_parse_heap();
+
+int main() {
+  int ret = 0;
+  atlogger_set_logging_level(ATLOGGER_LOGGING_LEVEL_INFO);
+
+  ret += test_1a_long_prompt();
+  ret += test_1b_short_prompt();
+  ret += test_1c_no_prompt();
+  ret += test_2a_no_token();
+  ret += test_3a_no_body();
+  ret += test_4a_empty_message();
+  ret += test_5a_heap();
+  ret += test_5b_bad_parse_heap();
+
+  if (ret > 0) {
+    atlogger_log("test_atserver_message", ATLOGGER_LOGGING_LEVEL_ERROR, "%d tests failed\n", ret);
+  }
+  return ret;
+}
+
+#define expect_uint(TEST_TAG, EXP, ACT)                                                                                \
+  if (EXP != ACT) {                                                                                                    \
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "%s: incorrect value (expected: %u, actual: %u)\n", TEST_TAG, EXP, \
+                 ACT);                                                                                                 \
+    return 1;                                                                                                          \
+  }
```

Comment:
macro for expecting two unsigned int types to be equal, otherwise log the error and return 1